### DISCUSSION TITLE
[Question] Cannot Train Humanoid AMP with Running Motion

### DISCUSSION THREAD

**Discussion by vuthanhcdt** [isaac-sim/IsaacLab#2403] [Q&A]
### Question

Hello,

I'm experimenting with the Humanoid AMP environment in skrl. When using motion data for "dance" and "walk", I was able to successfully train a policy that produced the desired behaviors. However, when switching to the "running" motion, the robot disappears from the scene even after 8000 training iterations. Upon inspection, I found that the actions returned were null.

I tried adjusting the Z-axis height offset from 0.15 to 0.25, but the issue persisted—the robot still disappears. I then modified the training parameters to include a learning rate scheduler:

```
learning_rate_scheduler: KLAdaptiveLR
learning_rate_scheduler_kwargs: 
  kl_threshold: 0.008
discriminator_loss_scale: 3.0
```

With these changes, training no longer crashes, but the resulting behavior does not resemble running. When the robot makes contact with the ground, the behavior appears unstable or abnormal. Could this be due to insufficient joint stiffness, or should I adjust certain parameters for training or modify the motion file itself?

I also increased the training duration, but the performance still falls short compared to Isaac Gym, where the running behavior could be learned in about 20 minutes.

Do you have any suggestions for resolving the agent disappearance issue or improving the training performance when using "running" motion?

Thanks in advance!

https://github.com/user-attachments/assets/d4973c7f-1bf1-4837-871f-5609872df5ee

**Comment by RandomOakForest** [comment]
Thank you for posting this question. That is an interesting outcome from the learning setup for running. I'll move this post to our _Discussions_ section for the team to follow up.
